BoxContentSDK/BoxContentSDK/BOXURLSessionManager.m

```diff
@@ -401,6 +401,48 @@ - (BOOL)cleanUpSessionTaskInfoGivenUserId:(NSString *)userId associateId:(NSStri
     return success;
 }
 
+- (void)cancelAndCleanUpBackgroundSessionTasksForUserId:(NSString *)userId error:(NSError **)outError
+{
+    //FIXME: make sure set up is completed, and prevents any new tasks created for this userId while we clean it up
+
+    if (userId != nil) {
+        NSError *error = nil;
+
+        //cancel on-going session tasks associated with this userId
+        [self cancelOnGoingSessionTasksForUserId:userId error:&error];
+        BOXAssert(error == nil, @"Failed to cancel on-going session tasks with error %@", error);
+
+        //clean up memory entries and cached info of all session tasks associated with this userId
+        NSArray *associateIds = [self.cacheClient associateIdsForUserId:userId error:&error];
```

Comment:
we could, but that only cleans up the cache dir, not in-memory info in self.backgroundSessionIdToSessionTask, so we just re-use the method that clean up all info relating to a session task